lectures/dynamic_programming/cake_eating_problem.md

```diff
@@ -0,0 +1,621 @@
+c^{-\gamma}$ falls faster with $c$.
+
+This suggests more smoothing, and hence a lower rate of consumption.
+
+In summary, we expect the rate of consumption to be *decreasing in both
+parameters*.
+
+Let's see if this is true.
+
+## The Value Function
+
+The first step of our dynamic programming treatment is to obtain the Bellman
+equation.
+
+The next step is to use it to calculate the solution.
+
+### The Bellman Equation
+
+To this end, we let $v(x)$ be maximum lifetime utility attainable from
+the current time when $x$ units of cake are left.
+
+That is,
+
+```{math}
+:label: value_fun
+
+v(x) = \max \sum_{t=0}^{\infty} \beta^t u(c_t)
+```
+
+where the maximization is over all paths $\{ c_t \}$ that are feasible
+from $x_0 = x$.
+
+At this point, we do not have an expression for $v$, but we can still
+make inferences about it.
+
+For example, as was the case with the {doc}`McCall model <../dynamic_programming/mccall_model>`, the
+value function will satisfy a version of the *Bellman equation*.
+
+In the present case, this equation states that $v$ satisfies
+
+```{math}
+:label: bellman-cep
+
+v(x) = \max_{0\leq c \leq x} \{u(c) + \beta v(x-c)\}
+\quad \text{for any given } x \geq 0.
+```
+
+The intuition here is essentially the same it was for the McCall model.
+
+Choosing $c$ optimally means trading off current vs future rewards.
+
+Current rewards from choice $c$ are just $u(c)$.
+
+Future rewards given current cake size $x$, measured from next period and
+assuming optimal behavior, are $v(x-c)$.
+
+These are the two terms on the right hand side of {eq}`bellman-cep`, after
+suitable discounting.
+
+If $c$ is chosen optimally using this trade off strategy, then we obtain maximal lifetime rewards from our current state $x$.
+
+Hence, $v(x)$ equals the right hand side of {eq}`bellman-cep`, as claimed.
+
+### An Analytical Solution
+
+It has been shown that, with $u$ as the CRRA utility function in
+{eq}`crra_utility`, the function
+
+```{math}
+:label: crra_vstar
+
+v^*(x_t) = \left( 1-\beta^{1/\gamma} \right)^{-\gamma}u(x_t)
+```
+
+solves the Bellman equation and hence is equal to the value function.
+
+You are asked to confirm that this is true in the exercises below.
+
+The solution {eq}`crra_vstar` depends heavily on the CRRA utility function.
+
+In fact, if we move away from CRRA utility, usually there is no analytical
+solution at all.
+
+In other words, beyond CRRA utility, we know that the value function still
+satisfies the Bellman equation, but we do not have a way of writing it
+explicitly, as a function of the state variable and the parameters.
+
+We will deal with that situation numerically when the time comes.
+
+Here is a Python representation of the value function:
+
+```{code-cell} julia
+function v_star(x, β, γ)
+
+    return (1 .- β .^ (1 ./ γ)) .^ (-γ) .* u(x, γ)
+end
+```
+
+And here's a figure showing the function for fixed parameters:
+
+```{code-cell} julia
+β, γ = 0.95, 1.2

```

Comment:
For this problem it is closest to the Python code. While technically correct, it might lead to confusion for the uninitiated on the named tuples. The next lecture has usage of named tuples.